src/Database/LSMTree/Internal.hs

```diff
@@ -512,9 +513,11 @@ lookups ks th fromEntry = withOpenTable th $ \thEnv -> do
     wb <- readMVar (tableWriteBuffer thEnv)
     let resolve = resolveMupsert (tableConfig th)
     cache <- readMVar (tableCache thEnv)
+    arenaManager <- newArenaManager -- TODO: use shared arena manager in handle (or session?)
```

Comment:
So here's my current thought (for use in future PRs):

Put the cache in the handle, but share the handle on `duplicate`. Only create a new independent cache in `new` and snapshot open.

This way independent tables will use independent caches and thus tables with different lifetimes will not affect the peak cache size in an extreme way. We'll not be able to be very aggressive with cache shrinking, if we have any shrinking at all.

Reply:
For the record: this sounds like we should not have independent tables in the single session to begin with. Then the cache will naturally be session-wide thing, instead of some intermediate thing between session and table-handle.